### PR TITLE
obsservability name changes

### PR DESCRIPTION
**Pull request by mounika-nalla**
### Description

Updated varibale names with logdna and sysdig to log_analysis and cloud_monitoring respectively.

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Changes made are as follows:

sysdig_instance -> cloud_monitoring_instance
sysdig_enabled -> cloud_monitoring_enabled
sysdig_instance_name -> cloud_monitoring_instance_name
sysdig_resource_group_id -> cloud_monitoring_resource_group_id
sysdig_access_key -> cloud_monitoring_access_key
sysdig_chart_location -> cloud_monitoring_chart_location
sysdig_agent_registry -> cloud_monitoring_agent_registry
sysdig_agent_namespace -> cloud_monitoring_agent_namespace
sysdig_key_validate_condition -> cloud_monitoring_key_validate_condition
sysdig_key_validate_msg -> cloud_monitoring_key_validate_msg
sysdig_key_validate_check -> cloud_monitoring_key_validate_check
sysdig_agent_tags -> cloud_monitoring_agent_tags
sysdig_agent -> cloud_monitoring_agent
sysdig_agent_version -> cloud_monitoring_agent_version
sysdig_metrics_filter -> cloud_monitoring_metrics_filter
sysdig_add_cluster_name -> cloud_monitoring_add_cluster_name
logdna_agent -> log_analysis_agent 
logdna-agent -> log-analysis-agent
logdna_instance_name -> log_analysis_instance_name
logdna_instance -> log_analysis_instance
logdna_agent_version -> log_analysis_agent_version
logdna_add_cluster_name -> log_analysis_add_cluster_name
logdna_ingestion_key -> log_analysis_ingestion_key
logdna_enabled -> log_analysis_enabled
logdna_agent_custom_line_exclusion -> log_analysis_agent_custom_line_exclusion
logdna_agent_custom_line_inclusion -> log_analysis_agent_custom_line_inclusion
logdna_agent_tags -> log_analysis_agent_tags
logdna_key_validate_check -> log_analysis_key_validate_check
logdna_key_validate_msg -> log_analysis_key_validate_msg
logdna_key_validate_condition -> log_analysis_key_validate_condition
logdna_agent_registry -> log_analysis_agent_registry
logdna_agent_namespace -> log_analysis_agent_namespace
logdna_resource_group_id -> log_analysis_resource_group_id
logdna_chart_location -> log_analysis_chart_location
logdna_secret_name -> log_analysis_secret_name

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,16 +78,16 @@ You can configure the logging agent to filter log lines according to the Kuberne
...
-custom_logdna_at_agent_line_inclusion = "namespace:default"
+custom_log_analysis_at_agent_line_exclusion = "label.app.kubernetes.io/name:sample-app\\, annotation.user:sample-user"
+custom_log_analysis_at_agent_line_inclusion = "namespace:default"
 ```
 
 The following is the corresponding DaemonSet configuration:
 
 ```text
-- name: LOGDNA_K8S_METADATA_LINE_INCLUSION
+- name: LOG_ANALYSIS_K8S_METADATA_LINE_INCLUSION
```
Comment (ocofaigh) [inline README.md:88]: This actually should be `LOGDNA_K8S_METADATA_LINE_INCLUSION` - please change it back

File: README.md [hunk 2]
```diff
@@ -78,16 +78,16 @@ You can configure the logging agent to filter log lines according to the Kuberne
...
   value: "label.app.kubernetes.io/name:sample-app, annotation.user:sample-user"
-- name: LOGDNA_K8S_METADATA_LINE_EXCLUSION
+- name: LOG_ANALYSIS_K8S_METADATA_LINE_EXCLUSION
```
Comment (ocofaigh) [inline README.md:90]: This actually should be `LOGDNA_K8S_METADATA_LINE_EXCLUSION` - please change it back

**Review by ocofaigh — CHANGES_REQUESTED** [review]
I think we went too far here. We should not be changing anything inside the charts. Just input variables and documentation

**Comment by ocofaigh** [comment]
Also this should not be a breaking change and we should not skip the upgrade test. Please include the text `UNSKIP UPGRADE TEST` in the next commit

**Comment by ocofaigh** [comment]
We probably need to use moved blocks to avoid this destroy:
```
Messages:   	Resource(s) identified to be destroyed 
        	            	Name: logdna_agent
        	            	Address: module.observability_agents.helm_release.logdna_agent[0]
        	            	Actions: [delete]
```

**Comment by mounika-nalla** [comment]
> We probably need to use moved blocks to avoid this destroy:
> 
> ```
> Messages:   	Resource(s) identified to be destroyed 
>         	            	Name: logdna_agent
>         	            	Address: module.observability_agents.helm_release.logdna_agent[0]
>         	            	Actions: [delete]
> ```

Despite having the moved.tf, upgrade tests failed in my local. I noticed that when I execute `terraform state mv` for the resources (logdna_agent, sysdig_agent), the upgrade tests passed. However, having the moved blocks didn't yield the same outcome.

**Review by toddgiguere — APPROVED** [review]

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**